google-cloud-firestore/src/main/java/com/google/cloud/firestore/AggregateField.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public abstract class AggregateField {
+  @Nonnull
+  public static CountAggregateField count() {
+    return new CountAggregateField();
+  }
+
+  @Nonnull
+  public static SumAggregateField sum(String field) {
+    return new SumAggregateField(FieldPath.fromDotSeparatedString(field));
+  }
+
+  @Nonnull
+  public static SumAggregateField sum(FieldPath fieldPath) {
+    return new SumAggregateField(fieldPath);
+  }
+
+  @Nonnull
+  public static AverageAggregateField average(String field) {
+    return new AverageAggregateField(FieldPath.fromDotSeparatedString(field));
+  }
+
+  @Nonnull
+  public static AverageAggregateField average(FieldPath fieldPath) {
+    return new AverageAggregateField(fieldPath);
+  }
+
+  @Nullable FieldPath fieldPath;
+
+  /** Returns the alias used internally for this aggregate field. */
+  String getAlias() {
+    // Use $operator_$field format if it's an aggregation of a specific field. For example: sum_foo.
+    // Use $operator format if there's no field. For example: count.
+    return getOperator() + (fieldPath == null ? "" : "_" + fieldPath.getEncodedPath());
```

Comment:
I don't think we should use `fieldPath.getEncodedPath()` here. The reason is that it will quote field paths with invalid characters. We actually want the entire alias to be quoted, not just the field path part.

if the field path is
```
contains`invalid
```

this code results in 
```
sum_`contains\`invalid`
```

but what we want is
```
`sum_contains\`invalid`
```

Reply:
Done